pkg/core/mem_pool.go

```diff
@@ -0,0 +1,250 @@
+package core
+
+import (
+	"sort"
+	"time"
+
+	"github.com/CityOfZion/neo-go/pkg/core/transaction"
+	"github.com/CityOfZion/neo-go/pkg/util"
+)
+
+// PoolItem represents a transaction in the the Memory pool.
+type PoolItem struct {
+	txn       *transaction.Transaction
+	timeStamp time.Time
+	fee       Feer
+}
+
+// PoolItems slice of PoolItem
+type PoolItems []*PoolItem
+
+// MemPool stores the unconfirms transactions.
+type MemPool struct {
+	unsortedTxn                 map[util.Uint256]*PoolItem
+	unverifiedTxn               map[util.Uint256]*PoolItem
+	sortedHighPrioTxn           PoolItems
+	sortedLowPrioTxn            PoolItems
+	unverifiedSortedHighPrioTxn PoolItems
+	unverifiedSortedLowPrioTxn  PoolItems
+
+	capacity int
+}
+
+func (p PoolItems) Len() int           { return len(p) }
+func (p PoolItems) Swap(i, j int)      { p[i], p[j] = p[j], p[i] }
+func (p PoolItems) Less(i, j int) bool { return (p[i].CompareTo(p[j]) < 0) }
```

Comment:
simplify:
```go
func (p PoolItems) Less(i, j int) bool { return p[i].CompareTo(p[j]) < 0 }
```

Reply:
You mean simplify `CompareTo`? The method `Less` seems to me already simply.

Reply:
compare `return (p[i].CompareTo(p[j]) < 0)` vs `return p[i].CompareTo(p[j]) < 0`

`( ... )` brackets not needed

Reply:
:+1: